com.unity.multiplayer.mlapi/Runtime/Core/RPCQueue/RPCQueueManager.cs

```diff
@@ -0,0 +1,549 @@
+        }
+
+        /// <summary>
+        /// AddToInternalMLAPISendQueue
+        /// NSS-TODO: This will need to be removed once we determine how we want to handle specific
+        /// internal MLAPI commands relative to RPCS.
+        /// Example: An network object is destroyed via server side (internal mlapi) command, but prior to this several RPCs are invoked for the to be destroyed object (Client RPC)
+        /// If both the DestroyObject internal mlapi command and the ClientRPCs are received in the same frame but the internal mlapi DestroyObject command is processed prior to the
+        /// RPCs being invoked then the object won't exist and additional warnings will be logged that the object no longer exists.
+        /// The vices versa scenario (create and then RPCs sent) is an unlikely/improbable scenario, but just in case added the CreateObject to this special case scenario.
+        ///
+        /// To avoid the DestroyObject scenario, the internal MLAPI commands (DestroyObject and CreateObject) are always invoked after RPCs.
+        /// </summary>
+        /// <param name="queueItem">item to add to the internal MLAPI queue</param>
+        public void AddToInternalMLAPISendQueue(FrameQueueItem queueItem)
+        {
+            rpcQueueProcessing?.QueueInternalMLAPICommand(queueItem);
+        }
+
+        /// <summary>
+        /// ProcessAndFlushRPCQueue
+        /// Will process the RPC queue and then move to the next available frame
+        /// </summary>
+        /// <param name="queueType"></param>
+        public void ProcessAndFlushRPCQueue(RPCQueueProcessingTypes queueType)
+        {
+            if (rpcQueueProcessing == null)
+            {
+                return;
+            }
+
+            switch (queueType)
+            {
+                case RPCQueueProcessingTypes.Receive:
+                {
+                    RPCQueueProcessing.ProcessReceiveQueue();
+                    break;
+                }
+                case RPCQueueProcessingTypes.Send:
+                {
+                    rpcQueueProcessing.ProcessSendQueue();
+                    break;
+                }
+            }
+        }
+
+        /// <summary>
+        /// GetCurrentFrame
+        /// Gets the current frame for the Inbound or Outbound queue
+        /// </summary>
+        /// <param name="qType"></param>
+        /// <returns>QueueHistoryFrame</returns>
+        public QueueHistoryFrame GetCurrentFrame(QueueHistoryFrame.QueueFrameType qType)
+        {
+            if (QueueHistory.ContainsKey(qType))
+            {
+                int StreamBufferIndex = GetStreamBufferIndex(qType);
+
+                if (QueueHistory[qType].ContainsKey(StreamBufferIndex))
+                {
+                    return QueueHistory[qType][StreamBufferIndex];
+                }
+            }
+
+            return null;
```

Comment:
is this an error condition?

Reply:
Well, if the method calling GetCurrentFrame finds that a returned null value is an error condition then it can throw it.
This means there are no QueueHistoryFrames instances (currently) for the QueueHistoryFrame.QueueFrameType specified.
The function calling this could use null as a way to determine if it needs to Initialize the queue or not.
null is "ok" to return when there are no objects to be returned.  :)